llitgi/ViewControllers/ManageTags/ManageTagsViewController.swift

```diff
@@ -0,0 +1,256 @@
+        return barButtonItem
+    }()
+    private lazy var loadingButton: UIBarButtonItem = {
+        let loading = UIActivityIndicatorView(style: .gray)
+        loading.startAnimating()
+        return UIBarButtonItem(customView: loading)
+    }()
+    
+    //MARK:- Class properties
+    let item: Item
+    let dataProvider: DataProvider
+    let themeManager: ThemeManager
+    let completed: () -> Void
+    private(set) var currentTags: [Tag] = []
+    private(set) var availableTags: [Tag] = []
+    
+    
+    //MARK:- Lifecycle
+    init(item: Item, dataProvider: DataProvider, themeManager: ThemeManager, completed: @escaping () -> Void) {
+        self.item = item
+        self.dataProvider = dataProvider
+        self.themeManager = themeManager
+        self.completed = completed
+        self.currentTags = item.tags
+        self.availableTags = dataProvider.tags.filter { tag in
+            return !item.tags.contains(where: { $0.name == tag.name} )
+        }
+        super.init(nibName: String(describing: ManageTagsViewController.self), bundle: Bundle(for: ManageTagsViewController.self))
+        self.title = item.title
+    }
+    
+    @available(*, unavailable)
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return self.themeManager.theme.statusBarStyle
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.navigationItem.leftBarButtonItem = self.cancelBarButtonItem
+        self.navigationItem.rightBarButtonItem = self.saveBarButtonItem
+        self.apply(self.themeManager.theme)
+        self.themeManager.addObserver(self) { [weak self] theme in
+            self?.apply(theme)
+        }
+        self.tableView.register(TagPickerCell.self)
+        self.tableView.dataSource = self
+        self.tableView.delegate = self
+        self.tableView.tableFooterView = UIView()
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        self.tableView.flashScrollIndicators()
+    }
+    
+    @objc
+    private func cancelTapped(_ sender: UIBarButtonItem) {
+        self.completed()
+    }
+    
+    @objc
+    private func saveTapped(_ sender: UIBarButtonItem) {
+        self.navigationItem.rightBarButtonItem = self.loadingButton
+        self.newTagBarButtonItem.isEnabled = false
+        self.tableView.isUserInteractionEnabled = false
+
+        let itemModification = ItemModification.init(action: .replaceTags(with: self.currentTags.map{ $0.name }), id: self.item.id)
+        self.dataProvider.performInMemoryWithoutResultType(endpoint: .modify(itemModification)) { [weak self] result in
+            guard let strongSelf = self else { return }
+            switch result {
+            case .isSuccess:
+                strongSelf.dataProvider.syncLibrary { _ in
```

Comment:
Should be enough modifying the tags here locally because the server already knows about the change. Not too bad for now, but definitely change it later on to avoid a useless sync operation.